### PR TITLE
add server/tools/repl.ts

### PR DESCRIPTION
**Pull request by McFlat**
### REPL ([Read Eval Print Loop](https://nodejs.org/docs/latest-v8.x/api/repl.html))

If you want to interact with the application libraries and objects, there is a REPL for that.

usage: `node ./dist/server/tools/repl.js`

"The default evaluator will, by default, assign the result of the most recently evaluated expression to the special variable `_` (underscore). Explicitly setting `_` to a value will disable this behavior."

- type `.help` to list commands available in the repl, notice it starts with a dot
- type `.exit` to exit, note that you still have to press CTRL-C to actually exit, or press CTRL-C (3 times) without typing `.exit` to exit
- type `context` to list all available objects and libraries in the context, note: `Promise` is also available but it's not listed in the context, in case you need promises for something
- type `env` to see the loaded environment variables
- type `path` to access path library
- type `lodash` to access lodash library
- type `uuidv1` to access uuid/v1 library
- type `uuidv3` to access uuid/v3 library
- type `uuidv4` to access uuid/v4 library
- type `uuidv5` to access uuid/v5 library
- type `YoutubeDL` to access youtube-dl library
- type `cli` to access the cli helpers object
- type `logger` to access the logger; if you log to it, it will write to stdout and to the peertube.log file
- type `constants` to access the constants loaded by the server
- type `coreUtils` to access the core-utils helpers object
- type `ffmpegUtils` to access the ffmpeg-utils helpers object
- type `peertubeCryptoUtils` to access the peertube-crypto helpers object
- type `signupUtils` to access the signup helpers object
- type `utils` to access the utils helpers object
- type `YoutubeDLUtils` to access the youtube-dl helpers object
- type `sequelizeTypescript` to access sequelizeTypescript
- type `modelsUtils` to access the models/utils
- type `models` to access the shortcut to sequelizeTypescript.models
- type `transaction` to access the shortcut to sequelizeTypescript.transaction
- type `query` to access the shortcut to sequelizeTypescript.query
- type `queryInterface` to access the shortcut to sequelizeTypescript.queryInterface

#### .help

```
PeerTube [1.0.0] (b10eb595)> .help
.break    Sometimes you get stuck, this gets you out
.clear    Break, and also clear the local context
.editor   Enter editor mode
.exit     Exit the repl
.help     Print this help message
.load     Load JS from a file into the REPL session
.r        Reset REPL
.reset    Reset REPL
.save     Save all evaluated commands in this REPL session to a file
PeerTube [1.0.0] (b10eb595)>
```

#### Lodash example

```
PeerTube [1.0.0] (b10eb595)> lodash.keys(context)
[ 'global',
  'console',
  'DTRACE_NET_SERVER_CONNECTION',
  'DTRACE_NET_STREAM_END',
  'DTRACE_HTTP_SERVER_REQUEST',
  'DTRACE_HTTP_SERVER_RESPONSE',
  'DTRACE_HTTP_CLIENT_REQUEST',
  'DTRACE_HTTP_CLIENT_RESPONSE',
  'process',
  'Buffer',
  'clearImmediate',
  'clearInterval',
  'clearTimeout',
  'setImmediate',
  'setInterval',
  'setTimeout',
  'XMLHttpRequest',
  'compact2string',
  'module',
  'require',
  'path',
  'repl',
  'context',
  'env',
  'lodash',
  'uuidv1',
  'uuidv3',
  'uuidv4',
  'uuidv5',
  'cli',
  'logger',
  'constants',
  'Sequelize',
  'sequelizeTypescript',
  'modelsUtils',
  'models',
  'transaction',
  'query',
  'queryInterface',
  'YoutubeDL',
  'coreUtils',
  'ffmpegUtils',
  'peertubeCryptoUtils',
  'signupUtils',
  'utils',
  'YoutubeDLUtils' ]
PeerTube [1.0.0] (b10eb595)>
```

#### YoutubeDL example
```
YoutubeDL.getInfo('https://www.youtube.com/watch?v=I5ZN289jjDo', function(err, data) {console.log(err, data)})
```

#### Models examples
```
PeerTube [1.0.0] (b10eb595)> new models.ActorModel({id: 3}).getVideoChannel().then(function(data){console.log(data.dataValues.name)})
Promise {
  _bitField: 0,
  _fulfillmentHandler0: undefined,
  _rejectionHandler0: undefined,
  _promise0: undefined,
  _receiver0: undefined }
PeerTube [1.0.0] (b10eb595)> Main root channel
PeerTube [1.0.0] (b10eb595)> let out; new models.UserModel({id: 1}).getAccount().then(function (data) {out = data.dataValues.id})
Promise {
  _bitField: 0,
  _fulfillmentHandler0: undefined,
  _rejectionHandler0: undefined,
  _promise0: undefined,
  _receiver0: undefined }
PeerTube [1.0.0] (b10eb595)> out
2
PeerTube [1.0.0] (b10eb595)>
```

Here is the issue I created before making this pull request: https://github.com/Chocobozzz/PeerTube/issues/1247

### CODE REVIEW

**Comment by Nutomic** [comment]
You should put a reference to the issue in your commit message, like `(fixes #1247)`.

**Comment by McFlat** [comment]
ok will do thanks @Nutomic

**Comment by Nutomic** [comment]
Usually the reference to the issue is put at the end of the commit message. You can check out other commits or PRs in this repo ;)

**Comment by McFlat** [comment]
I've squashed the commits for this pull request

**Comment by McFlat** [comment]
I still want to add utils and stuff to the repl

**Comment by McFlat** [comment]
Not sure if there's anything else to add, maybe it's ready to be merged, let me know

File: server/tools/repl.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import * as repl from 'repl'
+import * as path from 'path'
+import * as _ from 'lodash'
+import * as moment from 'moment'
```
[the remark this answers is not in the document]
Comment (McFlat) [reply server/tools/repl.ts]: okay, I can remove it no worries, my mistake

File: server/tools/repl.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import * as repl from 'repl'
```
[the remark this answers is not in the document]
Comment (McFlat) [reply server/tools/repl.ts]: I believe repl is part of node

**Comment by McFlat** [comment]
This is ready to be merged it you approve of this functionality. I'm not sure if we can add anything else to the repl, so for now it's good to go.

**Comment by McFlat** [comment]
I'm adding constants to the repl right now, gonna look for anything else

**Comment by McFlat** [comment]
I've pretty much added everything I can add, I can't see anything else that can be beneficial to having in the repl, it's ready to be merged

**Comment by McFlat** [comment]
Yep I will do that right now. What I'll do is paste the usage I wrote about it in the docs/tools

**Comment by rigelk** [comment]
Thansk @McFlat

**Comment by kazazes** [comment]
Has this been removed? If so, need to update docs.

**Comment by Chocobozzz** [comment]
@kazazes No it has not been removed, why?

**Pull request merged**